### DISCUSSION TITLE
How to pass custom params to resend callback url?

### DISCUSSION THREAD

**Discussion by michalkvasnicak** [nextauthjs/next-auth#11233] [Help]
I would like to pass some extra information when signing in using Resend in order to store that on user profile when it is first created.

So far I tried cookies but now help. I also tried following in my server action:

```ts
export async function signUp(formData) {
   return await signIn('resend', formData, { myCustomParam: '...' });
}
```

But the resulting callback URL in the email does not contain `myCustomParam` and there is no way to get it.

Is there any way how to do so?

**Comment by ramedov** [comment]
I know it's late, but I had the same question and for future googlers, you can pass the `email` directly like so:
```ts
await signIn('resend', {
    email: 'johndoe@example.com',
    redirectTo: '/lobby',
})
```

For some reason Typescript doesn't autocomplete the `email` property (at least on `v5-beta` version), but it's there; and passing it directly will allow to pass other params.
